### PR TITLE
Improvements on solvers API 

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Implements enhancements based on feedback by @mguidon and @newton1985  on the solvers API

- Refined ``File`` resource model: renamed class and fields; and changed required fields
- Applied more strictly [API design guidelines](https://cloud.google.com/apis/design)
     - Removed ``jobs`` resource  and added it as sub-collections of solver
     - Defines [Resource Names and Identifiers](https://cloud.google.com/apis/design/resource_names): ``name`` and ``id`` attributes are now reserved for resource name and id, resp. (see ``services/api-server/src/simcore_service_api_server/models/api_resources.py``)
- Simplified inputs/outputs models payloads: users set/get directly input/output objects in dictionaries with the name of the ports as keys
- This PR is paired with python-client's PR ITISFoundation/osparc-simcore-python-client#21

An overview of the new entrypoints

image

A sample with new API:

```python
    import os
    import time

    import osparc
    from osparc.models import File, Solver, Job, JobStatus, JobInputs, JobOutputs
    from osparc.api import FilesApi, SolversApi

    cfg = osparc.Configuration(
        host=os.environ.get("OSPARC_API_URL", "http://127.0.0.1:8006"),
        username=os.environ.get("MY_API_KEY"),
        password=os.environ.get("MY_API_SECRET"),
    )

    with osparc.ApiClient(cfg) as api_client:

        files_api = FilesApi(api_client)
        input_file: File = files_api.upload_file(file="path/to/input-file.h5")


        solvers_api = SolversApi(api_client)
        solver: Solver = solvers_api.get_solver_release("simcore/services/comp/isolve", "1.2.3")

        job: Job = solvers_api.create_job(
            solver.id,
            solver.version,
            JobInputs({"input_1": input_file, "input_2": 33, "input_3": False}),
        )

        status: JobStatus = solvers_api.start_job(solver.id, solver.version, job.id)
        while not status.stopped_at:
            time.sleep(3)
            status = solvers_api.inspect_job(solver.id, solver.version, job.id)
            print("Solver progress", f"{status.progress}/100", flush=True)

        outputs: JobOutputs = solvers_api.get_job_outputs(solver.id, solver.version, job.id)

        print( f"Job {outputs.job_id} got these results:")
        for output_name, result in outputs.results.items():
            print(output_name, "=", result)
```

## Question for reviewers:

1. do you find the example above self-explanatory?
1. can you adapt it to ``your-favorite-computational-service``  ? (i.e. set correct solver, i/o ...). Please provide it during your review.

## Related issue/s

- #1865 Run solvers from API
- Changes suggested by @mguidon @newton1985 

### Manually 
- make sure ``API_SERVER_DEV_FEATURES_ENABLED=1`` in `.env`
- build an deploy ``make build-devel up-devel``
- go to http://127.0.0.1:9081, create user and API token under ``Security-Settings>Create-API-Key``
- go to http://127.0.0.1:8006/dev/doc 
    -  Click upper left "Authorize" button and login using previous key/secret as username/pass
     - Test entries using "Try it out"
### Auto
```cmd
cd tests/public-api
make install-dev
make tests
```

## Checklist

- [x] REFACTOR: API model for file resource:  ``File`` and ``File.id, File.name``.
- [x] REFACTOR solver-API: Use ``File`` when job refers to a file at the input/output 
- [x] REFACTOR solver/job API: Avoid ``JobInput/Output`` and return directly types in the python client
- [x] FIX: Cannot list isolve-mpi
-  ~~Adds catalog-schema API models in models_library and create service fixture~~ (not until new catalog )
- [x] Merge PR client to master: https://github.com/ITISFoundation/osparc-simcore-python-client/pull/21
- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes

### CODE REVIEW

**Review by mguidon — COMMENTED** [review]
From an API perspective:
- I like how you changed the JobInputs/Outputs to contain Files now
- I do not like that you removed the JobsApi (i was only suggesting to move the job_creation from the the solvers api to the jobs api.

**Comment by pcrespov** [comment]
> * I do not like that you removed the JobsApi (i was only suggesting to move the job_creation from the the solvers api to the jobs api.) For instance, as a taskmanager, I want to get all the jobs in the system (or for a user or per service)

I understand your point (actually you mentioned to me earlier) but the design actually considers that extension (since jobs use global UUIDs). After carefully rereading the [API guidelines](https://cloud.google.com/apis/design/resources#resources), I decided include jobs resources as a sub-collection of solver resources. This change allows to configure specific attributes on ``jobs`` created by ``solvers``.  In the future, we will also have access points to any kind of jobs (regards-less of their creators/parents).

File: services/api-server/src/simcore_service_api_server/api/routes/solvers.py [hunk 1]
```diff
@@ -30,6 +24,8 @@
 # - TODO: :search? SEE https://cloud.google.com/apis/design/custom_methods#common_custom_methods
 # - TODO: move more of this logic to catalog service
 # - TODO: error handling!!!
+# - TODO: allow release_tags instead of versions in the next iteration.
+#    Would be nice to have /solvers/foo/releases/latest or solvers/foo/releases/3 , similar to docker tagging
```
Comment (mguidon) [inline services/api-server/src/simcore_service_api_server/api/routes/solvers.py:28]: Upvote

**Review by mguidon — COMMENTED** [review]
>  In the future, we will also have access points to any kind of jobs (regards-less of their creators/parents).
Ok, then I am happy.

**Review by mguidon — APPROVED** [review]
This looks good to me. (Except thati it is still no yet connected to the backend(
I will adapt my scripts to the current version and may or may not come back with feedback. But I think this looks mature enough.

**Review by sanderegg — APPROVED** [review]
I could try it after some fiddling thanks for the help.

it seems to work nicely. I find weird that one has to repeat the solver version everywhere but I can live with it.

File: services/api-server/src/simcore_service_api_server/api/routes/files.py [hunk 2]
```diff
@@ -8,15 +8,17 @@
 from uuid import UUID
 
 import httpx
-from fastapi import APIRouter, Depends, File, Header, UploadFile, status
+from fastapi import APIRouter, Depends
+from fastapi import File as FileParam
+from fastapi import Header, UploadFile, status
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/routes/files.py:13]: I guess

**Pull request merged**